src/core/ext/transport/chttp2/transport/writing.c

```diff
@@ -453,6 +453,7 @@ void grpc_chttp2_end_write(grpc_exec_ctx *exec_ctx, grpc_chttp2_transport *t,
 
   while (grpc_chttp2_list_pop_writing_stream(t, &s)) {
     if (s->sent_initial_metadata) {
+      s->write_initial_metadata_done = true;
```

Comment:
Is it possible for the initial metadata to be sent in multiple writes?  If so, we should probably set this when the first write finishes instead of waiting for all of the writes to finish.